tools/analyzer_plugin/lib/src/diagnostic/missing_required_prop.dart

```diff
@@ -220,7 +219,8 @@ class MissingRequiredPropDiagnostic extends ComponentUsageDiagnosticContributor
         continue;
       }
 
-      if (withNullability(result) || requiredness != PropRequiredness.late) {
+      // Late required prop validation is only enabled for props classes that have migrated to null safety.
+      if (propsClassElement.library.isNonNullableByDefault || requiredness != PropRequiredness.late) {
```

Comment:
This is the main fix; instead of checking `result.library`, we're now checking `propsClassElement.library`